src/Bicep.Core.UnitTests/Diagnostics/LinterRuleTests/UseSafeAccessRuleTests.cs

```diff
@@ -148,4 +149,95 @@ param roleAssignments roleAssignmentType
   roleDefinitionIdOrName: string
 }[]?
 """);
+
+    [TestMethod]
+    public void Codefix_recommends_safe_access_for_nullable_property_access() => AssertCodeFix("""
+param foo {
+  bar: string?
+}
+
+output bar string? = foo.b|ar
+""", """
+param foo {
+  bar: string?
+}
+
+output bar string? = foo.?bar
+""");
+
+    [TestMethod]
+    public void Rule_ignores_non_nullable_property_access() => AssertNoDiagnostics("""
+param foo {
+  bar: string
+}
+
+output bar string? = foo.bar
+""");
+
+    [TestMethod]
+    public void Codefix_recommends_safe_access_for_nullable_module_output_property_access() => AssertCodeFix("""
+module foo 'module.bicep' = {
+  name: 'foo'
+}
+
+output bar string? = foo.outputs.b|ar
+""", """
+module foo 'module.bicep' = {
+  name: 'foo'
+}
+
+output bar string? = foo.outputs.?bar
+""",
+supportingFiles: [new("module.bicep", """
+output bar string? = null
+""")]);
+
+    [TestMethod]
+    public void Codefix_recommends_safe_access_for_nullable_array_access() => AssertCodeFix("""
+param foo {
+  bar: string?
+}
+
+output bar string? = foo['b|ar']
+""", """
+param foo {
+  bar: string?
+}
+
+output bar string? = foo[?'bar']
+""");
+
+    [TestMethod]
+    public void Rule_ignores_non_nullable_array_access() => AssertNoDiagnostics("""
+param foo {
+  bar: string
+}
+
+output bar string? = foo['bar']
+""");
+
+    [TestMethod]
+    public void Rule_ignores_array_access_on_array() => AssertNoDiagnostics("""
+param foo (string | null)[]
+
+output bar string? = foo[0]
```

Comment:
Not for this PR, but in an ideal world, should we be suggesting `foo[?0]`? Out of bounds array accesses will throw, but we don't always have min/max length metadata on arrays (and we don't in this case).

Reply:
Good question. Might be worth creating an issue foi it?